src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs

```diff
@@ -12,15 +12,25 @@ public class ResolveRuntimePackAssets : TaskBase
     {
         public ITaskItem[] ResolvedRuntimePacks { get; set; }
 
+        public ITaskItem[] FrameworkReferences { get; set; } = Array.Empty<ITaskItem>();
+
         [Output]
         public ITaskItem[] RuntimePackAssets { get; set; }
 
         protected override void ExecuteCore()
         {
             List<TaskItem> runtimePackAssets = new List<TaskItem>();
```

Comment:
suggestion: extract some methods or even classes. This method is too long. Too many nested layer